### PR TITLE
[WCM] [Filesystem] Added documentation for tempnam

### PR DESCRIPTION
**Pull request by markchalloner**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | yes (symfony/symfony#14754) |
| Applies to | 2.7 |
| Fixed tickets | symfony/symfony#10018 |

### CODE REVIEW

File: components/filesystem/introduction.rst [hunk 1]
```diff
@@ -234,6 +234,20 @@ isAbsolutePath
     // return false
     $fs->isAbsolutePath('../dir');
 
+tempNam
```
Comment (xabbuh) [inline components/filesystem/introduction.rst]: should be `tempnam`

File: components/filesystem/introduction.rst [hunk 2]
```diff
@@ -234,6 +234,20 @@ isAbsolutePath
...
+~~~~~~~
+
+.. versionadded:: 2.7
+    The ``tempnam()`` function was introduced in Symfony 2.7.
+
+:method:`Symfony\\Component\\Filesystem\\Filesystem::tempnam` is a drop-in
+replacement for PHP's builtin tempnam which allows you to create a temporary
```
Comment (xabbuh) [inline components/filesystem/introduction.rst]: I would use the `phpfunction` role here to link to the documentation of the PHP `tempnam()` function:

```rst
[...] PHP's built-in :phpfunction:`tempnam` function which [...]
```

File: components/filesystem/introduction.rst [hunk 3]
```diff
@@ -234,6 +234,20 @@ isAbsolutePath
...
+file on a stream if the stream can be written to. Returns false on failure.
```
Comment (xabbuh) [inline components/filesystem/introduction.rst]: [...] if the stream is writable. It returns false on failure.

**Comment by stof** [comment]
This new method cannot be added in 2.7 (the PR was submitted after the feature freeze)

**Comment by markchalloner** [comment]
2.8 ok to rebase onto?

**Comment by markchalloner** [comment]
Rebased onto 2.8 in #5393

**Pull request closed without merging**